theano/tensor/nnet/tests/test_corr.py

```diff
@@ -453,13 +452,13 @@ def test_fwd(self):
 
         conv_unshared = corr.CorrMM(unshared=True, subsample=self.sub,
                                     filter_dilation=self.dil)(self.input, self.filters)
-        unshared_func = theano.function([self.input, self.filters], conv_unshared)
+        unshared_func = theano.function([self.input, self.filters], conv_unshared, mode=self.mode)
```

Comment:
This looks a bit weird, but it is because you're using the Python implementation as your reference, right? I'm thinking if it wouldn't be better to have a separate implementation for the tests, like the one that's used for the normal convolution.